File: problem_44.py
from math import sqrt, fabs

def ispentagonal(n):
    return ((1+sqrt(1+24*n))/6) == (int((1+sqrt(1+24*n))/6))

def pentagonal(n):
    return  n*(3*n-1)/2

run = True
n = 1
while(run):
    d = 1
    while(n-d > 0):
        if(ispentagonal(pentagonal(n)-pentagonal(n-d)) and ispentagonal(pentagonal(n)+pentagonal(n-d))):
            print(pentagonal(n), pentagonal(n-d), n)
            print(fabs(pentagonal(n)-pentagonal(n-d)))
            run = False
            break
        d+=1
    n+=1

File: problem_44.cpp
#include<bits/stdc++.h>
using namespace std;

#define ll long long 

int pentagonal_num(ll n){
  return n*(3*n - 1)/2;
}

bool ispentagonal(long double n){
  long double num = (1 + (double)(sqrt(24*n)))/6;
  long long num2 = (1+sqrt(24*n))/6;
  return num == num2;
}

void soln(){
  ll ans = 0;
  ll n=1;
  bool run=true;
  while(run){
	ll d = 1;
	while((n-d)>0){
	  if(ispentagonal(pentagonal_num(n)-pentagonal_num(n-d)) && ispentagonal(pentagonal_num(n)+pentagonal_num(n-d))){
		run = false;
		cout << abs(pentagonal_num(n)-pentagonal_num(n-d)) << "\n";
		break;
	  }
	  ++d;
	}
	cout << n << "\n";
	++n;
  }
}

int main(){
  soln();
  return 0;
}
